sdk/src/test/scala/com/horizen/certificatesubmitter/CertificateSubmitterTest.scala

```diff
@@ -995,10 +1018,12 @@ class CertificateSubmitterTest extends JUnitSuite with MockitoSugar {
     val mockedMainchainChannel: MainchainNodeChannel = mock[MainchainNodeChannel]
     val mockedSidechainNodeViewHolder = TestProbe()
 
-    val certificateSubmitterRef: TestActorRef[CertificateSubmitter] = TestActorRef(
-      Props(new CertificateSubmitter(mockedSettings, mockedSidechainNodeViewHolder.ref, mock[SecureEnclaveApiClient], params, mockedMainchainChannel)))
+    val keyRotationStrategy = new WithoutCircuitStrategy(mockedSettings, params, CryptoLibProvider.sigProofThresholdCircuitFunctions)
+
+    val certificateSubmitterRef: TestActorRef[CertificateSubmitter[CertificateDataWithoutKeyRotation]] = TestActorRef(
+      Props(new CertificateSubmitter(mockedSettings, mockedSidechainNodeViewHolder.ref, mock[SecureEnclaveApiClient], params, mockedMainchainChannel, keyRotationStrategy)))
 
-    val submitter: CertificateSubmitter = certificateSubmitterRef.underlyingActor
+    val submitter: CertificateSubmitter[CertificateDataWithoutKeyRotation] = certificateSubmitterRef.underlyingActor
 
```

Comment:
I don't think so, because in CertificateSubmitter we test general logic. Strategy-specific logic is covered with unit tests separately in WithKeyRotationCircuitStrategy and WithoutKeyRotationCircuitStrategy